### PR TITLE
Replace Jest with Vitest

### PR DESCRIPTION
**Pull request by longfeiwAmazonDE**
_Description of changes:_
Replace Jest with Vitest. This is to prepare wdio8 upgrade. [Wdio8](https://webdriver.io/blog/2022/12/01/webdriverio-v8-released/) also transitioned from Jest to Vitest. 

_Tests:_
```
longfeiw@b0de2892be74 wdio-aws-device-farm-service % npm run test

> wdio-aws-device-farm-service@7.1.2 test
> vitest run --silent


 RUN  v0.31.0 /Users/longfeiw/Documents/GitHub/wdio-aws-device-farm-service

 ✓ src/__tests__/launcher.test.ts (2)

 Test Files  1 passed (1)
      Tests  2 passed (2)
   Start at  10:19:15
   Duration  503ms (transform 43ms, setup 0ms, collect 191ms, tests 5ms, environment 0ms, prepare 55ms)
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: __mocks__/aws-sdk/clients/devicefarm.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (arnulfojr) [inline __mocks__/aws-sdk/clients/devicefarm.ts:1]: I believe we need to also update the runtime in which the tests are ran.
We need to include node16 and node18 - the tests running on node12 are having some warnings.

https://github.com/awslabs/wdio-aws-device-farm-service/blob/main/.github/workflows/test-lint.yml#L12

**Review by arnulfojr — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -61,12 +60,11 @@
     "eslint": "^8.27.0",
     "eslint-config-prettier": "^8.5.0",
     "husky": "^8.0.2",
-    "jest": "^27.2.1",
     "lint-staged": "^13.0.3",
     "prettier": "2.7.1",
-    "ts-jest": "^27.0.5",
     "ts-node": "^9.1.1",
-    "typescript": "^4.4.3"
+    "typescript": "^4.4.3",
```
Comment (ShradhaSangtani) [inline package.json:66]: Can you also upgrade typescript to 5.0.0

Comment (longfeiwAmazonDE) [reply package.json:66]: Doing that in the next PR with wdio8 upgrade.

File: .github/workflows/test-lint.yml [hunk 3]
```diff
@@ -4,24 +4,23 @@ on: [push, pull_request]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        node-version: [12.x, 14.x]
+        node-version: [16.x, 18.x]
```
Comment (arnulfojr) [inline .github/workflows/test-lint.yml:11]: And we are officially also dropping support for node14 💯

**Review by arnulfojr — APPROVED** [review]

**Pull request merged**